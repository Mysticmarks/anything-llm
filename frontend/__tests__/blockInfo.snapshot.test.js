const fs = require("fs");
const path = require("path");
const vm = require("vm");

describe("Agent builder block definitions", () => {
  it("includes website, file, and code blocks", () => {
    const filePath = path.resolve(
      __dirname,
      "../src/pages/Admin/AgentBuilder/BlockList/blockInfo.js"
    );
    const source = fs.readFileSync(filePath, "utf8");

    const transformed = source
      .replace(/^import[^;]+;\n/gm, "")
      .replace(
        /const ICONS = {[\s\S]*?};/,
        "const ICONS = new Proxy({}, { get: () => () => null });"
      )
      .replace(/export const /g, "const ")
      .concat(
        "\nmodule.exports = { BLOCK_TYPES, RAW_BLOCK_INFO, BLOCK_INFO };"
      );

    const sandbox = { module: { exports: {} }, exports: {} };
    vm.createContext(sandbox);
    new vm.Script(transformed, { filename: "blockInfo.js" }).runInContext(
      sandbox
    );

    const { BLOCK_TYPES, RAW_BLOCK_INFO } = sandbox.module.exports;

    expect(RAW_BLOCK_INFO[BLOCK_TYPES.WEBSITE]).toBeDefined();
    expect(RAW_BLOCK_INFO[BLOCK_TYPES.FILE]).toBeDefined();
    expect(RAW_BLOCK_INFO[BLOCK_TYPES.CODE]).toBeDefined();

    const snapshot = Object.fromEntries(
      Object.entries(RAW_BLOCK_INFO).map(([type, info]) => [
        type,
        {
          label: info.label,
          description: info.description,
          hasDefaultConfig: !!info.defaultConfig,
          summaryExample: info.getSummary(info.defaultConfig || {}),
        },
      ])
    );

    expect(snapshot).toMatchSnapshot();
  });
});
