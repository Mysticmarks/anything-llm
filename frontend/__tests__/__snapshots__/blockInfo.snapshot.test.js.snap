// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agent builder block definitions includes website, file, and code blocks 1`] = `
{
  "apiCall": {
    "description": "Make an HTTP request",
    "hasDefaultConfig": true,
    "label": "API Call",
    "summaryExample": "GET (no URL)",
  },
  "code": {
    "description": "Run custom code snippets",
    "hasDefaultConfig": true,
    "label": "Code Execution",
    "summaryExample": "Run javascript code",
  },
  "file": {
    "description": "Read from or write to local storage",
    "hasDefaultConfig": true,
    "label": "File",
    "summaryExample": "read (no path)",
  },
  "finish": {
    "description": "End of agent flow",
    "hasDefaultConfig": true,
    "label": "Flow Complete",
    "summaryExample": "Flow will end here",
  },
  "flowInfo": {
    "description": "Basic flow information",
    "hasDefaultConfig": true,
    "label": "Flow Information",
    "summaryExample": "Untitled Flow",
  },
  "llmInstruction": {
    "description": "Process data using LLM instructions",
    "hasDefaultConfig": true,
    "label": "LLM Instruction",
    "summaryExample": "No instruction",
  },
  "start": {
    "description": "Configure agent variables and settings",
    "hasDefaultConfig": false,
    "label": "Flow Variables",
    "summaryExample": "0 variables defined",
  },
  "webScraping": {
    "description": "Scrape content from a webpage",
    "hasDefaultConfig": true,
    "label": "Web Scraping",
    "summaryExample": "No URL specified",
  },
  "website": {
    "description": "Read or interact with a web page",
    "hasDefaultConfig": true,
    "label": "Open Website",
    "summaryExample": "read (no URL)",
  },
}
`;
